src/tests/e2etest/SettingsTests.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            public Regex PercentageDownloadThrottlePattern { get; set; }
+            public Regex CacheHostSourcePattern { get; set; }
+            public Regex CacheHostPattern { get; set; }
+            public Regex CacheHostFallbackPattern { get; set; }
+        }
+        public partial class ReportedConfiguration
+        {
+            public int value { get; set; } 
+        }
+
+        public partial class ReportedPolicies
+        {
+            public DeliveryOptimizationPolicies value { get; set; }
+        }
+        private int m_twinTimeoutSeconds;
+
+
+        public Settings GetSettingsReportObject()
+        {
+            Settings reportedObject = new Settings();
+            ReportedConfiguration reportedConfigurationObject = JsonSerializer.Deserialize<ReportedConfiguration>(GetNewTwin().Properties.Reported[ComponentName][ConfigurationProperty].ToString());
+            ReportedPolicies reportedPoliciesObject = JsonSerializer.Deserialize<ReportedPolicies>(GetTwin().Properties.Reported[ComponentName][PoliciesProperty].ToString());
+            reportedObject.DeviceHealthTelemetryConfiguration = reportedConfigurationObject.value;
+            reportedObject.DeliveryOptimizationPolicies = reportedPoliciesObject.value;
+            return reportedObject;
+        }
+
+        [SetUp]
+        public void TestSetUp()
+        {
+            m_twinTimeoutSeconds = base.twinTimeoutSeconds;
+            base.twinTimeoutSeconds = base.twinTimeoutSeconds * 2;
+        }
+
+        [TearDown]
+        public void TestTearDown()
+        {
+            base.twinTimeoutSeconds = m_twinTimeoutSeconds;
+        }
+
+        [Test]
+        public void SettingsTest_Set_Get()
+        {
+            var desiredDeliveryOptimizationPolicies = new DeliveryOptimizationPolicies
+            {
+                PercentageDownloadThrottle = 50,
+                CacheHostSource = 1,
+                CacheHost = "127.0.1.1",
+                CacheHostFallback = 90
+            };
+
+            var desiredSettings = new Settings
+            {
+                DeviceHealthTelemetryConfiguration = 2,
+                DeliveryOptimizationPolicies = desiredDeliveryOptimizationPolicies
+            };
+
+            if ((GetNewTwin().ConnectionState == DeviceConnectionState.Disconnected) || (GetTwin().Status == DeviceStatus.Disabled))
+            {
+                Assert.Fail("Module is disconnected or is disabled");
+            }
+
+            Twin twinPatch = CreateTwinPatch(ComponentName, desiredSettings);
+
+            var reportedObject = new Settings();
+            if (UpdateTwinBlockUntilUpdate(twinPatch))
+            {
+                reportedObject = GetSettingsReportObject();
+                // Wait until the reported properties are updated
+                DateTime startTime = DateTime.Now;
+                while(!IsSameByJson(desiredSettings, reportedObject) && (DateTime.Now - startTime).TotalSeconds < twinTimeoutSeconds)
```

Comment:
not understanding why we have to keep doing `IsSameByJson`? Once we set the setting, the next reported interval should contain the value. So all needed here is to wait, until the next update after the patch was applied, none of this json comparing is needed

Reply:
Thanks Ahmed. I have made all necessary changes as you requested.